frontend/packages/console-app/src/components/user-preferences/UserPreferenceCheckboxField.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { Checkbox, Skeleton } from '@patternfly/react-core';
+import { UserPreferenceCheckboxField as CheckboxFieldType } from '@console/dynamic-plugin-sdk/src';
+import { useUserSettings } from '@console/shared';
+import { UserPreferenceFieldProps } from './types';
+
+type UserPreferenceCheckboxFieldProps = UserPreferenceFieldProps<CheckboxFieldType>;
+
+const UserPreferenceCheckboxField: React.FC<UserPreferenceCheckboxFieldProps> = ({
+  id,
+  label,
+  userSettingsKey,
+  trueValue,
+  falseValue,
+}) => {
+  // resources and calls to hooks
+  const [
+    currentUserPreferenceValue,
+    setCurrentUserPreferenceValue,
+    currentUserPreferenceValueLoaded,
+  ] = useUserSettings<string>(userSettingsKey);
+
+  const loaded: boolean = currentUserPreferenceValueLoaded;
+
+  // utils and callbacks
+  const getCheckboxValue = (checked: boolean): string => (checked ? trueValue : falseValue);
+
+  const onChange = (checked) => {
+    const checkedValue: string = getCheckboxValue(checked);
+    checkedValue !== currentUserPreferenceValue && setCurrentUserPreferenceValue(checkedValue);
+  };
```

Comment:
Don't force that `trueValue` and `falseValue` are strings.

```suggestion
  // utils and callbacks
  const getCheckboxValue = (checked: boolean) => (checked ? trueValue : falseValue);

  const onChange = (checked) => {
    const checkedValue = getCheckboxValue(checked);
    checkedValue !== currentUserPreferenceValue && setCurrentUserPreferenceValue(checkedValue);
  };
```

Reply:
updated types